### ISSUE TITLE
(maybe) wrong memory context?

### ISSUE THREAD

**Issue by dvarrazzo** [eulerto/wal2json#10]
More a question than a bug report, as I don't know many details of Postgres allocator:

```c
pg_decode_change(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
				 Relation relation, ReorderBufferChange *change)

	/* ... */

	old = MemoryContextSwitchTo(data->context);

	/* ... */

	switch (change->action)
	{
		case REORDER_BUFFER_CHANGE_INSERT:
			if (change->data.tp.newtuple == NULL)
			{
				elog(WARNING, "no tuple data for INSERT in table \"%s\"", NameStr(class_form->relname));
				return;
			}
			break;

	/* ... */

	MemoryContextSwitchTo(old);
	MemoryContextReset(data->context);
}
```

if the return is early, isn't the memory context left in an inconsistent state?

Thank you.

**Comment by eulerto** [comment]
@dvarrazzo It leaks memory. It should release memory context before return.

**Comment by eulerto** [comment]
Fixed at commit ea2b35abad0793ffc53820f027a5ed61c774c99b
